Std/Data/BitVec/Lemmas.lean

```diff
@@ -122,11 +122,21 @@ private theorem lt_two_pow_of_le {x m n : Nat} (lt : x < 2 ^ m) (le : m ≤ n) :
 
 /-! ### cast -/
 
-@[simp] theorem toNat_cast (e : m = n) (x : BitVec m) : (cast e x).toNat = x.toNat := rfl
+section
+variable (h : w = v) (x : BitVec w)
```

Comment:
I would prefer *not* to use `variable` in a section in order to override explicitness. I know Mathlib does this, but it is hard to read and maintain. It's fine in the lemma statement to just write e.g. `theorem toNat_cast (h) : ...`

Reply:
Just `theorem toNat_cast (h) : ...` doesn't work, because Lean can't infer the second width of the equality. I just went with `theorem toNat_cast (h : w = v) (x : BitVec w) : ...` full-out everywhere